internal/seeder/seed.go

```diff
@@ -23,7 +25,52 @@ func NewSeedService(client *storage.Client, bucketId string, directoryRepo repo.
 	}
 }
 
+func newMetadata(obj *storage.ObjectAttrs) *model.Metadata {
+	return &model.Metadata{
+		Bucket:       obj.Bucket,
+		Name:         obj.Name,
+		Size:         obj.Size,
+		StorageClass: obj.StorageClass,
+		Created:      obj.Created,
+		Updated:      obj.Updated,
+	}
+
+}
+
+type objectIterator interface {
+	Next() (*storage.ObjectAttrs, error)
+}
+
 // Seed initiates the seeding process by traversing bucket and inserting into db
 func (s *SeedService) Start(ctx context.Context) error {
+	b := s.client.Bucket(s.bucketId)
+	if _, err := b.Attrs(ctx); err != nil {
+		return err
+	}
+
+	it := b.Objects(ctx, nil)
+	if err := s.insertFromIterator(it); err != nil {
+		return err
+	}
+	return nil
+}
+
+// insertFromIterator traverses iterator while inserting all containing items into db
+func (s *SeedService) insertFromIterator(it objectIterator) error {
+	for {
+		obj, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+
+		if err != nil {
+			return err
+		}
```

Comment:
combine these would save some compute cycles
```suggestion
		if err != nil {
			if err == iterator.Done {
			    break
		        }

			return err
		}
```

Reply:
also it would be better to wrap the error message with info about which step it failed, this will be helpful for later debugging